frontend/cypress/integration/dataSources.spec.js

```diff
@@ -56,4 +56,86 @@ describe('The Data Sources Page', () => {
+      cy.get('[data-testid="frozen-data-status"]').contains('Completed');
+
+      // Download frozen data source
+      cy.getAccessToken().then((token) => {
+        if (token) {
+          const options = {
+            url: `${Cypress.config('baseUrl')}/api/frozendata/`,
+            headers: {
+              Authorization: `token ${token.replaceAll('"', '')}`,
+            },
+          };
+          cy.request(options).then((response) => {
+            const currentDataSourceId = Math.max(...response.body.map((data) => Number(data.id)));
+            const currentFrozenDataSource = response.body.find(
+              (item) => Number(item.id) === currentDataSourceId,
+            );
+            cy.get('[data-testid="frozen-source-download-button"]')
+              .eq(0)
+              .should('not.be.visible')
+              .should(
+                'have.attr',
+                'href',
+                `/api/tables/download/${currentFrozenDataSource.frozen_db_table}/archives/`,
+              );
+          });
+        }
+      });
+
+      // Check if new frozen data source can  be accesses like others
+      cy.visit('/sources');
+      cy.get('[data-testid="sources-table-search"]').type('FTS ISO codes test{enter}');
+      cy.get('[data-testid="sources-table-row"]').first().contains('FTS ISO codes test');
+
+      // Query frozen data source
+      cy.visit('/queries/build');
+      cy.get('[name="name"]').focus().type('My Test Dataset');
+      cy.get('[name="description"]').focus().type('My Test Dataset Description');
+      cy.get('.search').eq(1).click({ force: true });
+      cy.wait(5000);
+      cy.get('.search').eq(1).type('FTS ISO codes test{enter}{esc}');
+      cy.get('[data-testid="qb-add-step-button"]').click();
+
+      cy.get('[name="name"]').eq(1).type('Dataset Step Test');
+      cy.get('[name="description"]').eq(1).type('Dataset Step Test Description');
+      cy.get('[data-testid="qb-step-select-query"]').type('select{enter}');
+      cy.get('.selectColumnCheckbox > input').eq(0).check({ force: true });
+      cy.get('[data-testid="qb-step-preview-button"]').click();
+      cy.get('[data-testid="qb-save-button"]').click();
+      cy.get('.dataset-row-footer').eq(0).contains('Frozen FTS ISO codes test')
+
+      // Delete new version
+      cy.visit('/sources');
+      cy.get('[data-testid="sources-table-search"]').type('FTS ISO codes{enter}');
+      cy.get('[data-testid="sources-table-row"]')
+        .first()
+        .then(($element) => {
+          cy.wrap($element).contains('Versions').click();
+          cy.get('[data-testid="frozen-data-description"]').first().contains('test');
+          cy.get('[data-testid="frozen-data-delete-button"]').first().dblclick({ force: true });
+        });
+
+      //  Confirm that frozen data source version was deleted
+      cy.visit('/sources');
+      cy.get('[data-testid="sources-table-search"]').type('FTS ISO codes test{enter}');
+      cy.get('.row').contains('No Data');
```

Comment:
Yeah, this is should be in, but before this, we'll need to know that the frozen record isn't showing up in the versions table ... I understand the previous challenge arose cause you were testing by name. So instead of that, verify that the number of frozen records is `X` in the beginning, and is `X - 1` after deleting.